app/main.py

```diff
@@ -1,214 +1,21 @@
-KNIGHTS = {
-    "lancelot": {
-        "name": "Lancelot",
-        "power": 35,
-        "hp": 100,
-        "armour": [],
-        "weapon": {
-            "name": "Metal Sword",
-            "power": 50,
-        },
-        "potion": None,
-    },
-    "arthur": {
-        "name": "Artur",
-        "power": 45,
-        "hp": 75,
-        "armour": [
-            {
-                "part": "helmet",
-                "protection": 15,
-            },
-            {
-                "part": "breastplate",
-                "protection": 20,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
-    }
-}
+from app.arena.preparation import Prepare
+from app.arena.fight import Battle
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def battle(knights: dict) -> dict:
+    fighters_list = []
+    battle_result = {}
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    for key, value in knights.items():
 
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
+        key = Prepare(value)
+        key.fighter_preparation()
+        fighters_list.append(key)
 
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
+    Battle.fight(fighters_list[0], fighters_list[2])
+    Battle.fight(fighters_list[1], fighters_list[3])
```

Comment:
Not so clear
Use dict instead of list

Reply:
don't understand how can I do it better with dict... no ideas... maybe hint?))

Reply:
key will be name of Knight and value obj of Knight
Using indexes for fight is worst solution